native/src/window/action.rs

```diff
@@ -35,16 +35,42 @@ pub enum Action<T> {
     SetMode(Mode),
     /// Sets the window to maximized or back
     ToggleMaximize,
+    /// Toggles whether window has decorations
+    /// ## Platform-specific
+    /// - **X11:** Not implemented.
+    /// - **iOS / Android / Web:** Unsupported.
+    ToggleDecorations,
+    /// Change whether or not the window will always be on top of other windows.
+    ///
+    /// ## Platform-specific
+    ///
+    /// - **iOS / Android / Web / Wayland:** Unsupported.
+    AlwaysOnTop(bool),
+    /// Sets the window icon.
+    ///
+    /// On Windows and X11, this is typically the small icon in the top-left
+    /// corner of the titlebar.
+    ///
+    /// ## Platform-specific
+    ///
+    /// - **iOS / Android / Web / Wayland / macOS:** Unsupported.
+    ///
+    /// - **Windows:** Sets `ICON_SMALL`. The base size for a window icon is 16x16, but it's
+    ///   recommended to account for screen scaling and pick a multiple of that, i.e. 32x32.
+    ///
+    /// - **X11:** Has no universal guidelines for icon sizes, so you're at the whims of the WM. That
+    ///   said, it's usually in the same ballpark as on Windows.
+    SetWindowIcon(I),
```

Comment:
This generic is polluting everything. We should have an `Icon` type that can go here and be converted to `winit::Icon`.

There is already an `Icon` in `iced`, maybe it can live in `iced_native` instead and hold the raw pixels `Vec<u8>` / width / height so it can be converted to the winit Icon outside this crate (iced_native doesn't / shouldn't depend on winit)

Reply:
```rs
// iced_native::window::icon

struct Icon {
    data: Vec<u8>,
    width: u32,
    height: u32,
}

// iced_winit::window

pub struct Icon(winit::window::Icon);

impl From<iced_native::window::Icon> for Icon {
    ..
}

impl From<Icon> for winit::window::Icon {
    ..
}

// iced_winit::settings

pub struct Window {
    icon: Option<crate::window::Icon>,
    ..
}

// iced::window
pub use iced_native::icon;
pub use iced_native::icon::Icon;
```

Reply:
ive done that still need to add some more functions but the main biggest issue is error checking.
```rs
impl From<iced_native::window::Icon> for Icon {
    fn from(value: iced_native::window::Icon) -> Self {
        let (rgba, width, height) = value.get_icon();
        Icon(winit::window::Icon::from_rgba(rgba, width, height).unwrap())
    }
}
```
that has no checks an winit icon has quite a few. so i could either write those checks onto iced_native::icon or figure out another way of doing it

Reply:
i will do some more work on it tm